dwl.c

```diff
@@ -328,9 +343,14 @@ buttonpress(struct wl_listener *listener, void *data)
 	case WLR_BUTTON_PRESSED:;
 		/* Change focus if the button was _pressed_ over a client */
 		if ((c = xytoclient(cursor->x, cursor->y))) {
-			surface = wlr_xdg_surface_surface_at(c->xdg_surface,
-					cursor->x - c->geom.x - c->bw,
-					cursor->y - c->geom.y - c->bw, NULL, NULL);
+			if (c->isxdg)
+				surface = wlr_xdg_surface_surface_at(c->xdg_surface,
+						cursor->x - c->geom.x - c->bw,
+						cursor->y - c->geom.y - c->bw, NULL, NULL);
+			else
+				surface = wlr_surface_surface_at(c->xwayland_surface->surface,
+						cursor->x - c->geom.x - c->bw,
+						cursor->y - c->geom.y - c->bw, NULL, NULL);
```

Comment:
Cases like these intentionally invoke different methods, as the xdg version has more functionality.